Zend/zend_alloc.h

```diff
@@ -26,14 +26,15 @@
 #include "../TSRM/TSRM.h"
 #include "zend.h"
 
-#ifndef ZEND_MM_ALIGNMENT
+#ifndef CONFIG_ZEND_MM_ALIGNMENT
 # define ZEND_MM_ALIGNMENT Z_UL(8)
-# define ZEND_MM_ALIGNMENT_LOG2 Z_L(3)
+# define ZEND_MM_ALIGNMENT_LOG2 Z_UL(3)
 #elif ZEND_MM_ALIGNMENT < 4
-# undef ZEND_MM_ALIGNMENT
-# undef ZEND_MM_ALIGNMENT_LOG2
 # define ZEND_MM_ALIGNMENT Z_UL(4)
-# define ZEND_MM_ALIGNMENT_LOG2 Z_L(2)
+# define ZEND_MM_ALIGNMENT_LOG2 Z_UL(2)
+#else
+# define ZEND_MM_ALIGNMENT Z_UL(CONFIG_ZEND_MM_ALIGNMENT)
+# define ZEND_MM_ALIGNMENT_LOG2 Z_UL(CONFIG_ZEND_MM_ALIGNMENT_LOG2)
```

Comment:
Shouldn't these be of type size_t rather than zend_ulong?

Reply:
They probably should, but `ZEND_MM_ALIGNMENT_LOG2` seemed to be used with hashkeys which are zend_ulong, I'll look some more into this.